src/nools/lib.js

```diff
@@ -140,40 +140,32 @@ function emitTasksForSchedule(c, schedule, r) {
   }
 }
 
-function emitContactBasedTargetFor(c, targetConfig) {
-  if(targetConfig.appliesIf && !targetConfig.appliesIf(c)) return;
+function emitTargetFor(targetConfig, c, r) {
+  var isEmittingForReport = !!r;
+  if(targetConfig.appliesIf && !targetConfig.appliesIf(c, r)) return;
 
-  var pass = !targetConfig.passesIf || !!targetConfig.passesIf(c);
+  var pass = !targetConfig.passesIf || !!targetConfig.passesIf(c, r);
+  var instanceDoc = isEmittingForReport ? r : c.contact;
+  var instance = createTargetInstance(targetConfig.id, instanceDoc, pass);
+
+  instance._id = (targetConfig.idType === 'report' ? r && r._id : c.contact._id) + '~' + targetConfig.id;
 
-  var instance = createTargetInstance(targetConfig.id, c.contact, pass);
   if(typeof targetConfig.date === 'function') {
-    instance.date = targetConfig.date(c);
+    instance.date = targetConfig.date(c, r);
   } else if(targetConfig.date === undefined || targetConfig.date === 'now') {
     instance.date = now.getTime();
   } else if(targetConfig.date === 'reported') {
-    instance.date = c.reported_date;
+    instance.date = isEmittingForReport ? r.reported_date : c.contact.reported_date;
```

Comment:
The old `c.reported_date` was incorrect? `c` is { contact, reports }?

Reply:
The old `c.reported_date` is undefined. `c.contact.reported_date` is what I believe was intended here.